docs/src/main/asciidoc/spring-di.adoc

```diff
@@ -1,42 +1,116 @@
-
-== Creating the Maven project
-
-First, we need a new project. Create a new project with the following command:
-
+endif::[]
+
+//downstream version of notes about GitHub links with downstream-only attributes
+ifdef::downstream[]
+[NOTE]
+====
+You can follow this guide and create the example that uses the {PRODUCTNAME} extension for the Spring Web API, or you can download and view the completed example.
+To view the completed {ProductName} Spring Web example, download it as an link:{URL_GITHUB_QUICKSTARTS_REPO_ARCHIVE}[archive] or clone the {PRODUCTNAME} examples Git link:{URL_GITHUB_QUICKSTARTS_REPO_BASE}.git[repository].
+You can find the Spring Web example in the link:{URL_GITHUB_QUICKSTARTS_DIRECTORY_BASE}spring-di-quickstart[`spring-di-quickstart` directory].
+====
+endif::[]
+
+// upstream tech preview status and notes
+ifdef::upstream[]
+include::./status-include.adoc[]
+:extension-status: preview
+endif::[]
+
+//tech preview info for downstream docs
+ifdef::downstream[]
+NOTE: The `quarkus-spring-di` extension is available as link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] with {ProductLongName}.
+endif::[]
+
+ifdef::downstream[]
+[IMPORTANT]
+====
+The Spring compatibility layer that {BrandName} provides does not start a Spring Application Context or execute any infrastructure classes that are provided by Spring (for example, `org.springframework.beans.factory.config.BeanPostProcessor` or `org.springframework.context.ApplicationContext`) when you start your application.
+{BrandName} can only read metadata from Spring classes and annotations and parse user code method return types and parameter types that are specific to Spring.
+However, when you add arbitrary libraries that are part of Spring Framework to your {BrandName} application, such libraries will not function properly, because {BrandName} is not able to use them.
+
+{BrandName} uses a Dependency Injection mechanism called ArC that is based on the https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html[Contexts and Dependency Injection for the Java 2.0 specification]. If you want to learn more about it we recommend you to read the https://quarkus.io/guides/cdi[{BrandName} introduction to CDI] and the https://quarkus.io/guides/cdi-reference#arc-configuration-reference[CDI reference guide].
+{BrandName} does not support test features provided by Spring Boot.
+To learn more about testing your {BrandName} code, see the https://quarkus.io/guides/getting-started-testing[{BrandName} testing guide].
+====
+endif::[]
+
+.Prerequisites
+
+* Have a {BrandName} Maven project.
+* Have OpenJDK {JDKVersion} installed.
+* Have Maven {maven-version} installed.
+
+ifdef::downstream[]
+[id="proc-creating-the-spring-di-example-maven-project_{context}"]
+endif::[]
+== Creating the Spring DI example Maven project
+
+ifdef::downstream[]
+[role="_abstract"]
+endif::[]
```

Comment:
Does this need to be conditionalized? The role thing is pretty harmless and won't do anything in community docs. I'm not a fan of these things, even though I understand why they are needed downstream. But I haven't found a way to reliably insert [role="_abstract"] into downstream adoc files because the line numbers aren't always the same. 

This is one of those things that might be kind of a trade-off and I think the community docs will have to learn to live with. It seems less invasive to me to just insert the tag and leave out the conditional statements. You could add a comment above it like:

----
//Downstream abstract tag. Please do not modify.
[role="_abstract"]
----

Reply:
Strictly speaking, no, it does not need to be condtionalized and I can just apply the solution that you are suggesting. let me do that.